### PR TITLE
added capability to take JSON as log's config

### PR DESCRIPTION
**Pull request by cuongtranx**
Added feature discussed #1775 

What changed is that I added the option to take in a JSON file for log's config using the `log-config-json` param.

### CODE REVIEW

**Comment by cuongtranx** [comment]
Thanks to take a look, I did squash the commits.

File: gunicorn/config.py [hunk 1]
```diff
@@ -1368,15 +1368,31 @@ class LogConfigDict(Setting):
...
+    cli = ["--log-config-json"]
+    meta = "FILE"
+    validator = validate_string
+    default = None
+    desc = """\
+    The log config JSON reads config from a JSON file
+
+    Format: https://docs.python.org/3/library/logging.config.html#logging.config.dictConfig
+
+    .. versionadded:: 19.9
```
Comment (berkerpeksag) [inline gunicorn/config.py]: I think "19.9" should be "20.0" now.

File: gunicorn/config.py [hunk 2]
```diff
@@ -1368,15 +1368,31 @@ class LogConfigDict(Setting):
...
 
+class LogConfigJson(Setting):
+    name = "logconfig_json"
+    section = "Logging"
+    cli = ["--log-config-json"]
+    meta = "FILE"
+    validator = validate_string
+    default = None
+    desc = """\
+    The log config JSON reads config from a JSON file
```
Comment (berkerpeksag) [inline gunicorn/config.py]: Perhaps using something shorter like "The log config to read from a JSON file." would be better?

File: gunicorn/config.py [hunk 3]
```diff
@@ -1368,15 +1368,31 @@ class LogConfigDict(Setting):
     desc = """\
     The log config dictionary to use, using the standard Python
     logging module's dictionary configuration format. This option
-    takes precedence over the :ref:`logconfig` option, which uses the
-    older file configuration format.
+    takes precedence over the :ref:`logconfig` and :ref:`logConfigJson` options, which uses the
```
Comment (berkerpeksag) [inline gunicorn/config.py]: Style nit: Please wrap lines at 80 chars.

File: gunicorn/config.py [hunk 4]
```diff
@@ -627,11 +627,11 @@ class WorkerClass(Setting):
         A string referring to one of the following bundled classes:
 
         * ``sync``
-        * ``eventlet`` - Requires eventlet >= 0.9.7 (or install it via 
+        * ``eventlet`` - Requires eventlet >= 0.9.7 (or install it via
```
Comment (berkerpeksag) [inline gunicorn/config.py]: I know that your editor did this automatically, but it would be nice not to include cosmetic fixes into this PR. Feel free to submit them as a separate PR.

File: gunicorn/glogging.py [hunk 5]
```diff
@@ -238,6 +239,21 @@ def setup(self, cfg):
...
             ) as exc:
                 raise RuntimeError(str(exc))
+        elif cfg.logconfig_json:
+            config = CONFIG_DEFAULTS.copy()
+            if os.path.exists(cfg.logconfig_json):
+                try:
+                    config_json = json.load(open(cfg.logconfig_json))
+                    config.update(config_json)
+                    dictConfig(config)
+                except(
```
Comment (berkerpeksag) [inline gunicorn/glogging.py]: Style nit: Add a space after `except`.

File: gunicorn/glogging.py [hunk 6]
```diff
@@ -238,6 +239,21 @@ def setup(self, cfg):
...
+                    json.JSONDecodeError,
+                    AttributeError,
+                    ImportError,
+                    ValueError,
+                    TypeError
+                ) as exc:
+                    raise RuntimeError(str(exc))
```
Comment (berkerpeksag) [inline gunicorn/glogging.py]: Would you like to submit a separate PR that changes this code to `raise RuntimeError(...) from None` after we merge this? There are a few more instances of this pattern, so it would be nice to leave it to a separate PR.

File: gunicorn/config.py (hunk 1 above)
Comment (cuongtranx) [reply gunicorn/config.py]: is there a plan to upgrade to 20.0? The current is 19.9 and this could be going to either 19.10 or 20.0.

File: gunicorn/glogging.py (hunk 6 above)
Comment (cuongtranx) [reply gunicorn/glogging.py]: I can certainly do that.

File: gunicorn/config.py (hunk 1 above)
Comment (berkerpeksag) [reply gunicorn/config.py]: The master branch doesn't support Python 2 anymore and since we can't drop Python 2 support in the next 19.x release, we can safely say that the master branch is now 20.0. If you want to update `gunicorn.__version__`, feel free to submit a separate PR :)

**Comment by ahsan-ul-haq** [comment]
@tilgovi @berkerpeksag Any thoughts on when this can get merged and released?

**Comment by BrettMoan** [comment]
@cuongtranx I today ran into an issue where i was basically trying to use `--log-config-dict` with a literal file path, which what this PR would enable via `--log-config-json`. (but obviously since it hasn't ever been merged I could not, infact, do that)

do you have any interest in continuing this PR?

**Comment by cuongtranx** [comment]
oh wow thanks for merging! This was a looong time ago and I didn't rmb putting this PR at all 😬
